src/__tests__/TaskclusterAuth/TaskclusterCallback.test.tsx

```diff
@@ -63,28 +61,34 @@ describe('Taskcluster Callback', () => {
       await screen.findByText(/Credentials were found/),
     ).toBeInTheDocument();
 
-    expect(window.fetch).toHaveBeenCalledWith(
+    expect(window.fetch).toHaveFetched(
       'https://firefox-ci-tc.services.mozilla.com/login/oauth/token',
       {
         method: 'POST',
-        body: expect.any(URLSearchParams) as URLSearchParams,
         headers: {
           'Content-Type': 'application/x-www-form-urlencoded',
         },
       },
     );
-    const requestBody = Object.fromEntries(
-      (window.fetch as jest.Mock).mock.calls[0][1].body as URLSearchParams,
-    );
 
+    const tokenCall = fetchMock.callHistory.lastCall(
+      'https://firefox-ci-tc.services.mozilla.com/login/oauth/token',
+    );
+    const body = tokenCall?.options.body;
+    if (!(body instanceof URLSearchParams)) {
+      throw new Error(
+        'Unexpected body passed to the endpoint /login/oauth/token',
+      );
+    }
+    const requestBody = Object.fromEntries(body);
```

Comment:
FYI this part is the less "automatic" part, where I had to change more code than needed. We don't have access to the direct calls now, but we can queries calls with the callHistory API https://www.wheresrhys.co.uk/fetch-mock/docs/API/CallHistory